Userland/Utilities/mkdir.cpp

```diff
@@ -8,27 +8,35 @@
 #include <AK/LexicalPath.h>
 #include <AK/StringBuilder.h>
 #include <LibCore/ArgsParser.h>
+#include <LibCore/System.h>
+#include <LibMain/Main.h>
 #include <errno.h>
 #include <sys/stat.h>
 #include <unistd.h>
 
-int main(int argc, char** argv)
+ErrorOr<int> serenity_main(Main::Arguments arguments)
 {
-    if (pledge("stdio cpath rpath", nullptr) < 0) {
-        perror("pledge");
-        return 1;
-    }
+    TRY(Core::System::pledge("stdio cpath rpath", nullptr));
 
     bool create_parents = false;
+    String mode_string;
     Vector<const char*> directories;
 
     Core::ArgsParser args_parser;
     args_parser.add_option(create_parents, "Create parent directories if they don't exist", "parents", 'p');
+    args_parser.add_option(mode_string, "Set new directory permissions", "mode", 'm', "octal-mode");
     args_parser.add_positional_argument(directories, "Directories to create", "directories");
-    args_parser.parse(argc, argv);
+    args_parser.parse(arguments);
 
-    // FIXME: Support -m/--mode option
-    mode_t mode = 0755;
+    mode_t default_mode = 0755;
+    mode_t mode = default_mode;
+
+    if (!mode_string.is_empty()) {
+        if (sscanf(mode_string.characters(), "%ho", &mode) != 1) {
```

Comment:
We have `AK::StringUtils::convert_to_uint_from_hex(..)` and friends, maybe we should have `AK::StringUtils::convert_to_uint_from_octal(..)`?

Reply:
Agreed, I had a peek into that module to circumvent the use of `sscanf` but couldn't find it. I'll look into it.